map_gui/src/load.rs

```diff
@@ -247,9 +258,30 @@ mod wasm_loader {
                     Ok(resp_value) => {
                         let resp: Response = resp_value.dyn_into().unwrap();
                         if resp.ok() {
-                            let buf = JsFuture::from(resp.array_buffer().unwrap()).await.unwrap();
-                            let array = js_sys::Uint8Array::new(&buf);
-                            tx.send(Ok(array.to_vec())).unwrap();
+                            let total_bytes = resp
```

Comment:
The error handling could be much more cautious. I fixed the few problems I encountered (like having more than 10 `got_read_bytes` messages queuing up).